elks/arch/i86/kernel/irq.c

```diff
@@ -58,19 +43,43 @@ static struct irqaction irq_action[] = {
 void do_IRQ(int i,void *regs)
 {
     register struct irqaction *irq = irq_action + i;
-
-    irq->handler(i, regs, irq->dev_id);
+    if (irq) irq->handler(i, regs, irq->dev_id);
```

Comment:
Add back in print "Unexpected interrupt: %d" in the case where irq == NULL. This is still useful for debugging device drivers.

Remove irq->dev_id, pass NULL instead. Always unused.

Reply:
Done in 990a917